### PR TITLE
[21726] Regression test for  Fast DDS-Gen Use different typename for collections of TK_UINT8 and TK_BYTE

### PR DESCRIPTION
**Pull request by richiware**
## Description

This PR adds a regression test for changes in generated coded coming from eprosima/fast-dds-gen#400 

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- *N/A* Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- _N/A_: If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesArraysDDSTypesTests.cpp [hunk 1]
```diff
@@ -588,6 +590,65 @@ TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_ArrayOctet)
     EXPECT_EQ(DynamicDataFactory::get_instance()->delete_data(data), RETCODE_OK);
 }
 
+void DDSTypesTest_ArrayUInt8_common()
+{
+    eprosima::fastdds::dds::DynamicTypesDDSTypesTest_DDSTypesTest_ArrayOctet_Test support;
```
Comment (MiguelCompany) [inline test/feature/dynamic_types/dds_types_tests/DynamicTypesArraysDDSTypesTests.cpp:595]: No need to instantiate the test, you could pass it by reference from the tests:

```suggestion
void DDSTypesTest_ArrayUInt8_common(
        eprosima::fastdds::dds::DynamicTypesDDSTypesTest& support)
{
```

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesArraysDDSTypesTests.cpp [hunk 2]
```diff
@@ -588,6 +590,65 @@ TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_ArrayOctet)
...
+}
+
+// Regression test for redmine ticket #20878.
+TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_ArrayUInt8_Regression20878)
+{
+    xtypes::TypeIdentifierPair regression_type_ids;
+    register_ArrayOctet_type_identifier(regression_type_ids);
+
+    DDSTypesTest_ArrayUInt8_common();
+}
```
Comment (MiguelCompany) [inline test/feature/dynamic_types/dds_types_tests/DynamicTypesArraysDDSTypesTests.cpp:650]: And then pass `*this`.

```suggestion
TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_ArrayUInt8)
{
    DDSTypesTest_ArrayUInt8_common(*this);
}

// Regression test for redmine ticket #20878.
TEST_F(DynamicTypesDDSTypesTest, DDSTypesTest_ArrayUInt8_Regression20878)
{
    xtypes::TypeIdentifierPair regression_type_ids;
    register_ArrayOctet_type_identifier(regression_type_ids);

    DDSTypesTest_ArrayUInt8_common(*this);
}
```

File: src/cpp/statistics/types/monitorservice_typesTypeObjectSupport.cxx [hunk 3]
```diff
@@ -734,7 +734,7 @@ void register_LivelinessChangedStatus_s_type_identifier(
             ReturnCode_t return_code_last_publication_handle {eprosima::fastdds::dds::RETCODE_OK};
             return_code_last_publication_handle =
                 eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->type_object_registry().get_type_identifiers(
-                "anonymous_array_uint8_t_16", type_ids_last_publication_handle);
+                "anonymous_array_byte_16", type_ids_last_publication_handle);
```
Comment (MiguelCompany) [inline src/cpp/statistics/types/monitorservice_typesTypeObjectSupport.cxx:737]: Will the changes in this file make the monitor service not to interoperate with previous versions?

File: src/cpp/statistics/types/typesTypeObjectSupport.cxx [hunk 4]
```diff
@@ -67,7 +67,7 @@ void register_EntityId_s_type_identifier(
             ReturnCode_t return_code_value {eprosima::fastdds::dds::RETCODE_OK};
             return_code_value =
                 eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->type_object_registry().get_type_identifiers(
-                "anonymous_array_uint8_t_4", type_ids_value);
+                "anonymous_array_byte_4", type_ids_value);
```
Comment (MiguelCompany) [inline src/cpp/statistics/types/typesTypeObjectSupport.cxx:70]: Same here

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/statistics/types/monitorservice_typesTypeObjectSupport.cxx (hunk 3 above)
Comment (richiware) [reply src/cpp/statistics/types/monitorservice_typesTypeObjectSupport.cxx:737]: This PR only modifies the name the type is registered under. This doesn't affect the interoperability.

File: src/cpp/statistics/types/typesTypeObjectSupport.cxx (hunk 4 above)
Comment (richiware) [reply src/cpp/statistics/types/typesTypeObjectSupport.cxx:70]: This PR only modifies the name the type is registered under. This doesn't affect the interoperability.

File: test/feature/dynamic_types/dds_types_tests/DynamicTypesArraysDDSTypesTests.cpp (hunk 2 above)
Comment (richiware) [reply test/feature/dynamic_types/dds_types_tests/DynamicTypesArraysDDSTypesTests.cpp]: Done in 19bb2db57.

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Pull request merged**